ConwaysGameOfLife.Tests/UniverseShould.cs

```diff
@@ -154,9 +154,113 @@ public void WhenTwoUniversesInstantiated_NotBeEqual()
             Cell cell = null;
             var universe = new Universe(3, 3);
 
-            var result = Assert.Throws<NullReferenceException>(() => universe.GetCellNeighbourLocations(cell));
+            Assert.Throws<NullReferenceException>(() => universe.GetCellNeighbourLocations(cell));
+            Assert.Throws<NullReferenceException>(() => universe.CountLiveNeighbours(cell));
+        }
+
+        [Fact]
+        public void WhenGivenANonBoundaryCell_ReturnCorrectNeighbourLocations()
+        {
+            var universe = new Universe(3, 3);
+            var cell = new Cell(CellState.Dead, 1, 1);
+            var expected = new List<Location>() {
+                new Location(0, 0), // x + 1, y + 1
+                new Location(0, 1),
+                new Location(0, 2),
+                new Location(1, 0),
+                new Location(1, 2),
+                new Location(2, 0),
+                new Location(2, 1),
+                new Location(2, 2)
+            };
             
-            Assert.Equal("Object reference not set to an instance of an object.", result.Message);
+            var result = universe.GetCellNeighbourLocations(cell);
+
+            Assert.True(UniverseHelper.ListsOfLocationsAreEqual(expected, result));
+        }
+
+        [Fact]
+        public void WhenGivenACellOnBoundaryX_ReturnCorrectNeighbourLocationsListInOrder()
+        {
+            var universe = new Universe(3, 3);
+            var cell = new Cell(CellState.Dead, 0, 1);
+            var expected = new List<Location>() {
+                new Location(2, 0),
+                new Location(2, 1),
+                new Location(2, 2),
+                new Location(0, 0),
+                new Location(0, 2),
+                new Location(1, 0),
+                new Location(1, 1),
+                new Location(1, 2)
+            };
+
+            var result = universe.GetCellNeighbourLocations(cell);
+
+            Assert.True(UniverseHelper.ListsOfLocationsAreEqual(expected, result));
+        }
+
+        [Fact]
+        public void WhenGivenABoundaryCellX_ReturnCorrectNeighbourLocationsListOutOfOrder()

```

Comment:
what is the purpose of this test compared to the previous one? Is it required?

Reply:
Sorry I may have suggested that it's okay to have the expected list in a different order without clarifying that the `UniverseHelper.ListsOfLocationsAreEqual` should already handle when the lists aren't in the same order